### DISCUSSION TITLE
remove js comments in inline bundle

### DISCUSSION THREAD

**Discussion by electriquo** [parcel-bundler/parcel#9395] [Q&A] [answered]
how can i make parcel minify all inline content by removing unnecessary spaces in html and comments in js files?
this is what i have and what i tried.

index.html
```html
<html>
  <head>
  </head>
  <body>
    <script type="module">
      import "./index.js";
    </script>
  </body>
</html>
```

index.js
```js
// index.js
/* index.js */
```

```shell
$ jq -r '.devDependencies.parcel' package.json
2.10.3

$ npx parcel build  --public-url .
✨ Built in 98ms

dist/index.html    113 B    267ms

$ cat dist/index.html
<html><head><style></style></head><body> <script type="module">// index.js
/* index.js */</script> </body></html>
```

**Comment by electriquo** [comment, marked as answer]
found the solution:
for removing comments in js files https://github.com/parcel-bundler/parcel/issues/7563#issuecomment-1204042776
for removing comments in html files https://htmlnano.netlify.app/modules#collapsewhitespace

```shell
$ cat .htmlnanorc
{
  "collapseWhitespace": "aggressive"
}

$ cat .terserrc
{
  "format": {
    "comments": false
  }
}
```

**Reply by mischnic** [reply]
The terserrc should ideally not be necessary, there was a bug recently in swc that required this. But it will not be required anymore once the next swc version is published.

**Reply by electriquo** [reply]
> next swc version is published

what is the link for swc to keep an eye for a release?

**Reply by mischnic** [reply]
It's in swc v1.3.99 which was released today

**Reply by electriquo** [reply]
https://github.com/swc-project/swc

**Reply by mischnic** [reply]
I probably should have said specifically: bump the version of `@swc/core` in your lockfile to 1.3.99, or newer. Then remove `.parcel-cache` and rebuild

**Reply by electriquo** [reply]
`npm update` updated @swc/core to 1.3.99. yet, looking at the [version changes](https://github.com/swc-project/swc/compare/v1.3.99...main) i do not see anything related and
> removing unnecessary spaces in html

still persist. having .htmlnanorc as above, solves this and seems necessary.

**Reply by mischnic** [reply]
On https://github.com/swc-project/swc/compare/v1.3.97...v1.3.99 ,  there's https://github.com/swc-project/swc/pull/8257 listed which says
> This PR also fixes the `Default` implementation of `JsMinifyFormatOptions`

that should make `"comments": false` the default again.

In a freshly created project and your source code I get this output:
```html
<html><head></head><body> <script type="module"></script> </body></html>;
```

**Reply by electriquo** [reply]
thank you for the reference and looking into it deeper.
observe the spaces surrounding `<script>...</script>` and i assume the `;` at the end is a typo.

**Reply by mischnic** [reply]
You should ask/report an issue for htmlnano why this isn't the default behaviour.

This issue seems related: https://github.com/posthtml/htmlnano/issues/34

And indeed,
```html
<html>
  <head>
    <style>body{color: red;}</style>
    <script type="module">import "./index.js";</script>
  </head>
  <body></body>
</html>
```
(so putting them in the head) is minified to
```html
<html><head><style>body{color:red}</style><script type="module"></script></head><body></body></html>;
```

So there might be a reason why it leaves the whitespace inside `<body>`.

**Reply by electriquo** [reply]
thank you
